articles/understanding-systemd/index.md

```diff
@@ -0,0 +1,291 @@
+
+systemd also provides a tool called "***systemctl***". ***man 1 systemctl*** says ***systemctl*** may be used to introspect and control the state of the "***systemd***" system and service manager.
+
+This tool can be used to start and stop services including other things as we will soon see. System V init implementations manage system state by the use of shell scripts. 
+systemd on the flip-side takes a different approach towards writing configuration files. It does this by expressing commands in a declarative format. This makes the job of administering Linux systems easier by making uniform all aspects of system administration.
+
+
+[ULSAH](https://www.amazon.com/UNIX-Linux-System-Administration-Handbook/dp/0134277554) gives a succinct description on systemd's mode of operation:
+
+*Much like a package management system, systemd defines a robust dependency model, not only among services but also among "targets," systemd's term for the operational modes that traditional init calls runlevels. systemd not only manages processes in parallel, but also manages network connections (networkd), kernel log entries (journald), and logins (logind).*
+
+systemd is also compatible with existing init scripts of System V.
+
+---
+
+### systemd deep-dive
+During boot time systemd loads its configuration, determines its boot goal and dependences. It then activates them afterward.
+
+systemd also doesn't follow a sequential pattern for starting services. Rather, a flexible process by which extra components activate when system events occur.
+
+The systemd Linux manual states that systemd provides a dependency system between various entities called "units".
+Units encapsulate various objects that are relevant for system boot-up and maintenance. A **unit type** is a type of task relevant to the management of a Linux system. A task is a unit in systemd parlance. These units are a standardized representation of system resources managed by the suite of systemd daemons. 
+
+There are default unit files supplied by the packages of the service that usually reside in **/lib/systemd/system/**.  These files are vendor-written and can be overwritten by the package maintainer in the event of a package update. User-specific configurations for units are usually made in **/etc/systemd/system/** because the unit files in this location won't be overwritten in the event of a package upgrade.
+
+**Note:** It's common practice to copy the default unit file into **/etc/systemd/system/** and make appropriate customization.
+
+There are 11 type of units in systemd. We'd talk about the most common ones. ***man 1 systemd*** describes them. So give it a check.
+
+1. Service units: This unit defines how to manage service daemons controlled by systemd. Unit files of this service end with .service.
+2. Socket units: This contains information about a local inter-process communication or network sockets in a Linux system.
```

Comment:
Misuse of an article  '......about a local....' should be '..... about local....'